File: versioned_docs/version-1.9.x/examples/rs/examples/access_control_browse_tokens.rs
use reduct_rs::{Permissions, QuotaType, ReductClient, ReductError};
use tokio;

#[tokio::main]
async fn main() -> Result<(), ReductError> {
    // Create a new client with the API URL and API token
    let client = ReductClient::builder()
        .url("http://127.0.0.1:8383")
        .api_token("my-token")
        .build();

    // Browse all tokens and print their information
    let tokens = client.list_tokens().await?;
    for token in tokens {
        println!("Token: {}", token.name);
        println!("Created: {:?}", token.created_at);
        println!("Provisioned: {:?}", token.is_provisioned);

        // Get detailed information about the token
        let token = client.get_token(&token.name).await?;
        println!("Permissions: {:?}", token.permissions);
    }

    Ok(())
}

File: versioned_docs/version-1.19.x/examples/go/src/access_control_browse_tokens.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
)

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})
	// Browse all tokens and print their information
	tokens, err := client.GetTokens(context.Background())
	if err != nil {
		panic(err)

	}

	for _, token := range tokens {
		println("Token:", token.Name)
		println("Created:", token.CreatedAt)
		println("Provisioned:", token.IsProvisioned)

		// Get detailed information about a specific token
		details, err := client.GetToken(context.Background(), token.Name)
		if err != nil {
			panic(err)
		}
		println("Permissions:", details.Permissions)
	}
}
